node/src/accountant/mod.rs

```diff
@@ -2769,6 +3754,724 @@ pub mod tests {
+               make the transaction paid",
+        );
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "tx receipt for pending '0x0000…007b' - tx status: code other than 0 or 1 shouldn't be possible, but was 456"
+    )]
+    fn check_out_transaction_receipt_panics_at_undefined_status_code() {
+        let mut tx_receipt = TransactionReceipt::default();
+        tx_receipt.status = Some(U64::from(456));
+        let mut payment_backup = make_payment_backup();
+        payment_backup.hash = H256::from_uint(&U256::from(123));
+        let subject = AccountantBuilder::default().build();
+
+        let _ = subject.check_out_transaction_receipt(
+            tx_receipt,
+            payment_backup,
+            &Logger::new("receipt_check_logger"),
+        );
+    }
+
+    #[test]
+    fn is_non_pending_is_properly_set() {
+        assert_eq!(
+            PendingTransactionStatus::Failure(make_transaction_id()).is_non_pending(),
+            true
+        );
+        assert_eq!(
+            PendingTransactionStatus::Confirmed(make_payment_backup()).is_non_pending(),
+            true
+        );
+        assert_eq!(
+            PendingTransactionStatus::StillPending(make_transaction_id()).is_non_pending(),
+            false
+        )
+    }
+
+    #[test]
+    fn accountant_handles_payment_backup() {
+        init_test_logging();
+        let insert_payment_backup_params_arc = Arc::new(Mutex::new(vec![]));
+        let pending_payment_dao_factory = PendingPaymentsDaoFactoryMock::new(
+            PendingPaymentsDaoMock::default()
+                .insert_payment_backup_params(&insert_payment_backup_params_arc)
+                .insert_payment_backup_result(Ok(())),
+        );
+        let subject = AccountantBuilder::default()
+            .pending_payments_dao_factory(Box::new(pending_payment_dao_factory))
+            .build();
+        let accountant_addr = subject.start();
+        let tx_hash = H256::from_uint(&U256::from(55));
+        let accountant_subs = Accountant::make_subs_from(&accountant_addr);
+        let amount = 4055;
+        let timestamp = SystemTime::now();
+        let backup_message = PaymentBackupRecord {
+            rowid: 0,
+            timestamp,
+            hash: tx_hash,
+            attempt: 0,
+            amount,
+            process_error: None,
+        };
+
+        let _ = accountant_subs
+            .payment_backup
+            .try_send(backup_message.clone())
+            .unwrap();
+
+        let system = System::new("ordering payment backup test");
+        System::current().stop();
+        assert_eq!(system.run(), 0);
+        let insert_payment_backup_params = insert_payment_backup_params_arc.lock().unwrap();
+        assert_eq!(
+            *insert_payment_backup_params,
+            vec![(tx_hash, amount, timestamp)]
+        );
+        TestLogHandler::new().exists_log_containing(
+            "DEBUG: Accountant: Processed a backup for payment '0x0000…0037'",
```

Comment:
If we're going to be searching the blockchain for this wallet, it'd be nice to have the whole address in the log, rather than the abbreviated version.